nexus/src/app/instance_platform.rs

```diff
@@ -492,3 +494,108 @@ impl super::Nexus {
+//
+// This is a free function (and not an `Into` impl on `VmmCpuPlatform`) to keep
+// all of the gnarly CPUID details out of the DB model crate, which defines that
+// type.
+fn cpuid_from_vmm_cpu_platform(
+    platform: db::model::VmmCpuPlatform,
+) -> Option<Cpuid> {
+    macro_rules! cpuid_leaf {
+        ($leaf:literal, $eax:literal, $ebx:literal, $ecx:literal, $edx:literal) => {
+            CpuidEntry {
+                leaf: $leaf,
+                subleaf: None,
+                eax: $eax,
+                ebx: $ebx,
+                ecx: $ecx,
+                edx: $edx,
+            }
+        };
+    }
+
+    macro_rules! cpuid_subleaf {
+        ($leaf:literal, $sl:literal, $eax:literal, $ebx:literal, $ecx:literal, $edx:literal) => {
+            CpuidEntry {
+                leaf: $leaf,
+                subleaf: Some($sl),
+                eax: $eax,
+                ebx: $ebx,
+                ecx: $ecx,
+                edx: $edx,
+            }
+        };
+    }
+
+    // See [RFD 314](https://314.rfd.oxide.computer/) section 6 for all the
+    // gnarly details.
+    const MILAN_CPUID: [CpuidEntry; 32] = [
+        cpuid_leaf!(0x0, 0x0000000D, 0x68747541, 0x444D4163, 0x69746E65),
+        cpuid_leaf!(0x1, 0x00A00F11, 0x00000800, 0xF6F83203, 0x078BFBFF),
+        cpuid_leaf!(0x5, 0x00000000, 0x00000000, 0x00000000, 0x00000000),
+        cpuid_leaf!(0x6, 0x00000002, 0x00000000, 0x00000000, 0x00000000),
+        cpuid_subleaf!(
+            0x7, 0x0, 0x00000000, 0x219C03A9, 0x00000000, 0x00000000
+        ),
+        cpuid_subleaf!(
+            0xB, 0x0, 0x00000001, 0x00000002, 0x00000100, 0x00000000
+        ),
+        cpuid_subleaf!(
+            0xB, 0x1, 0x00000000, 0x00000000, 0x00000201, 0x00000000
+        ),
+        cpuid_subleaf!(
+            0xD, 0x0, 0x00000007, 0x00000340, 0x00000340, 0x00000000
+        ),
+        cpuid_subleaf!(
+            0xD, 0x1, 0x00000007, 0x00000340, 0x00000000, 0x00000000
+        ),
+        cpuid_subleaf!(
+            0xD, 0x2, 0x00000100, 0x00000240, 0x00000000, 0x00000000
+        ),
+        cpuid_leaf!(0x80000000, 0x80000021, 0x68747541, 0x444D4163, 0x69746E65),
+        cpuid_leaf!(0x80000001, 0x00A00F11, 0x40000000, 0x444001F0, 0x27D3FBFF),
+        cpuid_leaf!(0x80000002, 0x73736F72, 0x726F6365, 0x31332050, 0x43203737),
+        cpuid_leaf!(0x80000003, 0x20455059, 0x00414D44, 0x00000000, 0x00000000),
+        cpuid_leaf!(0x80000004, 0x00000000, 0x00000000, 0x00000000, 0x00000000),
+        cpuid_leaf!(0x80000005, 0xFF40FF40, 0xFF40FF40, 0x20080140, 0x20080140),
+        cpuid_leaf!(0x80000006, 0x08002200, 0x68004200, 0x02006140, 0x01009140),
+        cpuid_leaf!(0x80000007, 0x00000000, 0x00000000, 0x00000000, 0x00000100),
+        cpuid_leaf!(0x80000008, 0x00003030, 0x111ED205, 0x00000000, 0x00000000),
+        cpuid_leaf!(0x8000000A, 0x00000000, 0x00000000, 0x00000000, 0x00000000),
+        cpuid_leaf!(0x80000019, 0xF040F040, 0xF040F040, 0x00000000, 0x00000000),
```

Comment:
guests in fact see all zeroes between 0x8000000B and 0x80000019.

for 0x80000019, a guest currently sees:

> `0x80000019 0x00: eax=0xf040f040 ebx=0xf0400000 ecx=0x00000000 edx=0x00000000`

which is the same as hardware, and differs in `ebx`'s low word. that is the "L2 instruction TLB associativity/number of entries" for 1G pages, so i assume the duplication is on the expectation that the L2 TLB is the same for data/instructions, but in fact is only described for data? i'll fix this one to match hardware and update RFD 314 accordingly.